worker/src/error.rs

```diff
@@ -1,19 +1,70 @@
+use crate::Response;
+use serde_json::json;
+use std::result::Result;
+use thiserror::Error;
 use wasm_bindgen::{JsCast, JsValue};
 
 /// All possible Error variants that might be encountered while working with a Worker.
-#[derive(Debug)]
+
+#[derive(Error, Debug)]
 #[non_exhaustive]
 pub enum Error {
     BadEncoding,
     BodyUsed,
+    ResponseError(Response),
     Json((String, u16)),
     JsError(String),
     Internal(JsValue),
     BindingError(String),
-    RouteInsertError(matchit::InsertError),
+    RouteInsertError(#[from] matchit::InsertError),
     RouteNoDataError,
-    RustError(String),
-    SerdeJsonError(serde_json::Error),
+    ParseError(#[from] url::ParseError),
+    StringError(String),
+    SerdeJsonError(#[from] serde_json::Error),
+    RustError(#[from] anyhow::Error),
+}
+
+#[inline]
+pub fn unwrap_abort<T, E>(o: Result<T, E>) -> T {
```

Comment:
I don't see why this needs to exist, I think just regular `expect` calls would be better. I might not be understanding correctly though, why this instead of an `expect`?